### ISSUE TITLE
ant test fails but Travis doesn't notice

### ISSUE THREAD

**Issue by regisd** [jflex-de/jflex#431]
The examples running ant are failing.
See #384  #429 
However, Travis didn't block the submit.

**Comment by regisd** [comment]
Still the same problem with incorrect error codes.

e.g. on `examples/simple`:
```
ant test
```

>   [junit] Test YylexTest FAILED
> BUILD SUCCESSFUL
> Total time: 8 seconds

```
echo $?
```

> 0

**Comment by regisd** [comment]
The ant `<exec>` task has `failonerror`. It's quite annoying and error-prone if all tasks have to be annotated like this. https://ant.apache.org/manual/Tasks/exec.html

**Comment by lsf37** [comment]
Will try to clean up the ant examples for the release.

**Comment by lsf37** [comment]
For posterity: it looks like this is specific to the junit task, the other tasks seem to generate non-zero exit codes as expected.

The junit task counts a run with failed tests as an error-free run. It would fail in the shell sense if there was some kind of unexpected exception. The workaround to make failing tests fail the junit task is a bit annoying, but seems to do what we need:

```
<target name="test" depends="build">
    <junit failureproperty="test.failed">
     ... 
    </junit>
    <fail message="Test failure." if="test.failed"/>
</target>
 ```

**Comment by lsf37** [comment]
It looks like
```
<target name="test" depends="build">
    <junit haltonfailure="yes">
     ... 
    </junit>
</target>
```
also has the desired effect, and is a bit shorter.

**Comment by lsf37** [comment]
Right, and not just the junit task, but also the java task, as you said. javac, cup, and jflex behave as expected.

**Comment by lsf37** [comment]
This should now be addressed in a5475cb9a6e0e914fa1ebaea1cd07bd6e2cb4569 and #709.
